coins/monero/src/wallet/scan.rs

```diff
@@ -128,6 +128,10 @@ impl ReceivedOutput {
     self.data.commitment.clone()
   }
 
+  pub fn arbitrary_data(&self) -> Vec<Vec<u8>> {
```

Comment:
I'd return &[Vec<u8>] here. That lets the user iterate over it without a clone, or if they need an owned instance, clone it themselves.